File: MovieRaterApi/Data/Entities/UserMovie.cs
namespace MovieRaterApi.Data.Entities;

public class UserMovie
{
    public Guid UserId { get; set; }
    public Guid MovieId { get; set; }
    public bool IsFavorite { get; set; }
    public bool IsInWatchlist { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public User User { get; set; } = null!;
    public Movie Movie { get; set; } = null!;
}

File: movie-rater-fe/src/types/user-movie.ts
export interface UserMovieResponse {
  userId: string
  movieId: string
  isFavorite: boolean
  isInWatchlist: boolean
  createdAt: string
  updatedAt: string
}

export interface UserMovieWithMovie {
  id: string
  tmdbId: number
  title: string
  posterUrl: string | null
  backdropUrl: string | null
  releaseDate: string | null
  voteAverage: number
  isFavorite: boolean
  isInWatchlist: boolean
  createdAt: string
  updatedAt: string
}

export interface PagedUserMoviesResponse {
  page: number
  totalPages: number
  totalResults: number
  results: UserMovieWithMovie[]
}

export interface UserMovieListRequest {
  favoritesOnly?: boolean
  watchlistOnly?: boolean
  page?: number
  pageSize?: number
}